src/test/java/homework_2/random_chars_table/RandomCharsTableTest.java

```diff
@@ -0,0 +1,133 @@
+package homework_2.random_chars_table;
+
+import base.UnitBase;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.Test;
+
+import java.util.NoSuchElementException;
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class RandomCharsTableTest extends UnitBase {
+    @Test
+    void InputZeroAsArgument() {
+        setInput("0 0 odd");
+
+        new RandomCharsTable().run();
+        printOut();
+        removeFromOutput("Введите данные:");
+        assertEquals("Passed parameters should match the format [positive integer] [positive integer] [even|odd]", getOutputLines()[0]);
+    }
+
+    @Test
+    void InputEmptyArgument() {
+        setInput("");
+        assertThrows(NoSuchElementException.class, () -> new RandomCharsTable().run());
+    }
+
+    /*
+        for this test use new RandomCharsTable().setDebug(true).run();
+        3 3 odd
+        | K | N | W |
+        | N | E | E |
+        | J | G | D |
+        Even letters - N, N, J, D
+     */
+    @Test
+    void InputValidEvenArguments() {
+        setInput("3 3 even");
+
+        new RandomCharsTable().setDebug(true).run();
+        printOut();
+        removeFromOutput("Введите данные:");
+        assertEquals("| K | N | W |", getOutputLines()[0]);
+        assertEquals("| N | E | E |", getOutputLines()[1]);
+        assertEquals("| J | G | D |", getOutputLines()[2]);
+        assertEquals("Even letters - N, N, J, D", getOutputLines()[3]);
+    }
+
+    /*
+        for this test use new RandomCharsTable().setDebug(true).run();
+        3 3 odd
+        | K | N | W |
+        | N | E | E |
+        | J | G | D |
+        odd letters - K, W, E, E, G
+     */
+    @Test
+    void InputValidOddArguments() {
+        setInput("3 3 odd");
+
+        new RandomCharsTable().setDebug(true).run();
+        printOut();
+        removeFromOutput("Введите данные:");
+        assertEquals("| K | N | W |", getOutputLines()[0]);
+        assertEquals("| N | E | E |", getOutputLines()[1]);
+        assertEquals("| J | G | D |", getOutputLines()[2]);
+        assertEquals("Odd letters - K, W, E, E, G", getOutputLines()[3]);
+    }
+
+    @Test
+    void InputRandomRowsAndColumnsAsArgument() {
+        final String STRATEGY = "even";
+        final int ROWS = 1 + new Random().nextInt(20);
+        final int COLUMNS = 1 + new Random().nextInt(20);
+        final String input = "" + ROWS + " " + COLUMNS + " " + STRATEGY;
+
+        setInput(input);
+
+        new RandomCharsTable().run();
+        printOut();
+        removeFromOutput("Введите данные:");
+        for (int i = 0; i < ROWS; i++)
+        assertTrue(getOutputLines()[i].matches("(\\|\\s[A-Z]\\s){" + COLUMNS + "}\\|"));
+    }
+
+    // дальше тест уродлив, но как иначе распарсить эти строки и сравнить все я не знаю
+    // за то покрывает все варианты.
+    @RepeatedTest(10)
+    void InputRandomCheckStrategy() {
```

Comment:
It seems it does the trick, but possible bugs could be present. If possible, simplify :)